src/mesh/coordinates.cxx

```diff
@@ -679,41 +679,40 @@ const Field3D Coordinates::Vpar_Grad_par(const Field3D &v, const Field3D &f, CEL
 const Field2D Coordinates::Div_par(const Field2D &f, CELL_LOC outloc,
                                    DIFF_METHOD method) {
   TRACE("Coordinates::Div_par( Field2D )");
-  // Perversely the relevant coordinates object might not be `this` if
-  // f isn't at the same location.
-  if (f.getLocation() != location)
-    return f.getCoordinates()->Div_par(f, outloc, method);
 
-  return Bxy * Grad_par(f / Bxy, outloc, method);
+  // Need Bxy at location of f, which might be different from location of this
+  // Coordinates object
+  Field2D Bxy_floc = f.getCoordinates()->Bxy;
+
+  return Bxy * Grad_par(f / Bxy_floc, outloc, method);
```

Comment:
Implicit assumption that `location == outloc` (or at least that `location ==` location of return from `Grad_par`), should we assert this (maybe level 1?) to make it explicit?

Reply:
:+1: I've added checks through all the Coordinates derivatives/difops now.